src/crimson/os/seastore/onode_manager/staged-fltree/stages/node_stage_layout.cc

```diff
@@ -36,7 +36,7 @@ void F013_T::update_size_at(
     NodeExtentMutable& mut, const me_t& node, index_t index, int change)
 {
   assert(index <= node.num_keys);
-  extent_len_t node_size = mut.get_length();
+  [[maybe_unused]] extent_len_t node_size = mut.get_length();
```

Comment:
I remember Kefu has a PR to remove [[maybe_unused]] in the code, since there is warning in some certain build environment "warning: maybe_unused attribute ignored" , and add COMPILE_FLAGS " -Wno-unused-variable" in CMakeLists.txt.

Reply:
What is the preferred way to suppress this warning instead? is it `std::ignore`?

Reply:
Kefu just add COMPILE_FLAGS " -Wno-unused-variable" in CMakeLists.txt.

Reply:
Could you paste the related PR?

Reply:
https://github.com/ceph/ceph/pull/42787

Reply:
hmm, OK, I'll drop this commit in the next push.

Reply:
i think #42787 was a special case, where the `maybe_unused` variable was a member variable. @rzarzynski removed it to silence the warning from GCC-10. in this case, `node_size` is a local variable. if both GCC-10 and Clang are happy with the `[[maybe_unused]]` specifier, i think that'd be a better way than passing `-Wno-unused-variable` to the compiler.

Reply:
Yeah, adding `[[maybe_unused]]` is better as it is not a member variable.